src/components/FriendshipButton/FriendshipButton.tsx

```diff
@@ -36,13 +39,21 @@ const FriendshipButton = (props: Props) => {
     ? friendAddress.slice(0, isTableAndBelow ? MOBILE_ADDRESS_SIZE : DESKTOP_ADDRESS_SIZE)
     : getAvatarName(profile?.avatars[0]).fullName
 
+  const handleAddFriend = () => {
+    if (isAuthDappEnabled && !isLoggedIn) {
+      window.location.replace(`${config.get('AUTH_URL')}?redirectTo=${window.location.href}`)
+    } else {
+      onAddFriend()
+    }
+  }
+
   const handleButtonClick = useCallback(() => {
     switch (friendshipStatus) {
       case FriendshipStatus.FRIEND:
         setIsOpenModal(true)
         break
       case FriendshipStatus.NOT_FRIEND:
-        return onAddFriend()
+        return handleAddFriend()
```

Comment:
If I'm not wrong, we need to add the `handleAddFriend`, `setIsOpenModal` and `onAcceptFriendRequest` methods to the callback dependencies.